### ISSUE TITLE
Payment Sep 25 [$1000] Workspaces - Visual flickering when navigating back to workspace list from a WS

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#24341]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open the staging web link
2. Go to Settings > Workspaces
3. Select a workspace to open WS settings
4. Go back to workspace list by tapping on the back button

## Expected Result:
No visual issues / screen should slide smoothly

## Actual Result:
Visual flickering on the button "New workspace" when going back to workspace list

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [x] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.52.1

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/4af40d02-b0ae-4e8d-b2d2-041c8045206c

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01cbc57cc70a4d58dc</li>
        <li>Upwork Job ID: 1689518567778115584</li>
        <li>Last Price Increase: 2023-08-24</li>
    </ul>
</details>

**Comment by sophiepintoraetz** [comment]
I'm able to reproduce this and I think this is a design polish (but necessary)

https://github.com/Expensify/App/assets/42940078/5772b110-802b-44e8-b140-9ce03ae3d92a

**Comment by swakeert** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
On ios/ Safari, when navigating back to the Workspace list from a particular workspaces' setting, there is a visual flickering and occasional movement of "New Workspace" text.

### What is the root cause of that problem?
On moving between the Workspace list and Workspace Setting page, there is a change in the safe area background. When the Workspace setting page is unmounted, it is causing a slightly delayed re-render of the footer and safe area which is causing the 'flickering' effect.

### What changes do you think we should make in order to solve the problem?
We need to inspect the state and find the root cause of the re-render. And ensure it does not perform the re-render when coming back from the workspace setting page.

### What alternative solutions did you explore? (Optional)
I checked the transition effects on the footer and headerPageLayout and its child components to see what could have caused the re-render, but did not find anything.

**Comment by Thanos30** [comment]
I personally can't replicate it in my local environment, so that makes it hard to test. I found this which might be useful:

https://github.com/react-navigation/react-navigation/issues/9530

**Comment by sophiepintoraetz** [comment]
@allroundexperts - any thoughts on the proposal here?

**Comment by Thanos30** [comment]
@allroundexperts any idea why this isn't reproducable on my localhost?

**Comment by swakeert** [comment]
I'll need to spend some time with the code and fiddle around. Once I have the answer, it would probably be a few lines of code anyway. Not sure what more I can suggest without actually solving the problem.

**Comment by Thanos30** [comment]
It happens on bot live websites, it doesn't happen when running locally.

**Comment by swakeert** [comment]
@allroundexperts in that case I'll try to fix it locally and raise a PR or send a code diff. But since I'm not officially assigned this task yet, I'll have to keep it lower on my personal priority. :)

**Comment by Thanos30** [comment]
@swakeert were you able to reproduce the issue locally on your environment by any chance? Without that being possible I couldn't provide a solution.

**Comment by Thanos30** [comment]
@allroundexperts Got it, thanks, I will re-check

**Comment by studentofcoding** [comment]
# Proposal

## Please re-state the problem that we are trying to solve in this issue.

Visual flickering when navigating back to workspace list from a WS

## What is the root cause of that problem?

The root of the problem is Smart App Banner that only loads on `WorkspaceListPage` and not in `WorkspaceSettingPage`, thus making the initial layout change suddenly. This is only happening on iOS, as Chrome didn't have Native Smart App Banner integration

1. WorkspaceListPage

<img width="529" alt="Screenshot 2023-08-15 at 03 16 44" src="https://github.com/Expensify/App/assets/20473526/dabdf783-65c7-4521-80c7-a6ecb235d447">

2. WorkspaceSettingPage
<img width="529" alt="Screenshot 2023-08-15 at 03 17 06" src="https://github.com/Expensify/App/assets/20473526/4086de74-e220-4946-bf65-99cc9c328f6d">


## What changes do you think we should make in order to solve the problem?

We need to ensure that the Smart App Banner also loads on `WorkspaceSettingPage` so the layout is not changing, or hide it altogether.

## What alternative solutions did you explore? (Optional)

We can also enforce styling to accommodate this sudden change, but it's a workaround

cc: @allroundexperts

**Comment by swakeert** [comment]
@allroundexperts sure, sounds good.

@Thanos30 It is reproducible on my iPhone on prod. Not exactly like the video, where there is a resize of the button, but the background does flicker for me. I can't reproduce it on my local because I haven't set up access to my local development web server from my phone.

FYI: The issue is not reproducible using chrome dev tools, or similar. I believe the source of the issue is ios-safe-area used for safari on iPhone.

**Comment by studentofcoding** [comment]
@allroundexperts Updating my proposal with a Demo of the problem didn't occur on the web if we didn't have the NewDot App installed

**Comment by sophiepintoraetz** [comment]
Waiting on a proposal review from @allroundexperts

**Comment by studentofcoding** [comment]
Sure @allroundexperts, So upon more exploration, the Core is really on the Smart App Banner that suddenly shows on the Workspaces list, and I still suggest that we use the **Add Smart App Banner [configuration](https://developer.apple.com/documentation/webkit/promoting_apps_with_smart_app_banners#3701576)** as with this solution, our OLD Smart App Banner will be replaced by New Configured Smart App Banner, thus
1. It'll be shown all over the app, and not break any structure
2. User can close it, and it'll not show again after that.

But we can see how the CSS Workaround works below

## CSS Workaround
When Smart App Banner is just shown suddenly & breaks the structure of components for a split second, what we can do is make the flickering not really seen by adding animating it, via adding styles of 

```
autoResize: {
        height: '100svh',
        minHeight: '64px',
        overflow: 'hidden',
        transition: 'height 0.5s ease',
    },
 ```
 
 Then we can add this style into the `IllustratedHeaderPageLayout` on `ScreenWrapper` component below
 ```
 <ScreenWrapper
            style={[StyleUtils.getBackgroundColorStyle(backgroundColor),styles.autoResize]}
            shouldEnablePickerAvoiding={false}
            includeSafeAreaPaddingBottom={false}
            offlineIndicatorStyle={[StyleUtils.getBackgroundColorStyle(themeColors.appBG)]}
        >
            {({safeAreaPaddingBottomStyle}) => (
                <>
                    <HeaderWithBackButton
                        // eslint-disable-next-line react/jsx-props-no-spreading
                        {...propsToPassToHeader}
                        titleColor={backgroundColor === themeColors.appBG ? undefined : themeColors.textColorfulBackground}
                        iconFill={backgroundColor === themeColors.appBG ? undefined : themeColors.iconColorfulBackground}
                    />
                    <View style={[styles.flex1, StyleUtils.getBackgroundColorStyle(themeColors.appBG)]}>
                        <ScrollView
                            contentContainerStyle={safeAreaPaddingBottomStyle}
                            showsVerticalScrollIndicator={false}
                        >
                            <View style={styles.overscrollSpacer(backgroundColor, windowHeight)} />
                            <View style={[styles.alignItemsCenter, styles.justifyContentEnd, StyleUtils.getBackgroundColorStyle(backgroundColor)]}>
                                <Lottie
                                    source={illustration}
                                    style={styles.w100}
                                    autoPlay
                                    loop
                                />
                            </View>
                            <View style={[styles.pt5]}>{children}</View>
                        </ScrollView>
                        {!_.isNull(footer) && <FixedFooter>{footer}</FixedFooter>}
                    </View>
                </>
            )}
        </ScreenWrapper>
 ```
 
 Result on new.expensify.com
  
https://github.com/Expensify/App/assets/20473526/4ac2558b-dba1-4602-af0a-fdd42a39ff2f

## Add Smart App Banner Configuration 

Below is the Demo of the Implementation when it Loaded, & Loaded & closed

- Loaded

https://github.com/Expensify/App/assets/20473526/98fe21a7-6193-4709-8566-aa0233ec6a66


- Loaded & then Closed

https://github.com/Expensify/App/assets/20473526/f3e0232c-ee31-4534-b540-27f2b46b8300

**Comment by sophiepintoraetz** [comment]
Still reviewing proposals!

**Comment by studentofcoding** [comment]
So upon deeper research, we have set `universal_links` with several URL structures on this file `apple-app-site-association` file

https://github.com/Expensify/App/blob/df510ff83101d045972e8eaf3607b4e923d9b578/.well-known/apple-app-site-association#L6C1-L66

> **What is universal_links?** 
> Universal linking is Apple's method of launching apps on iOS when linked from a website.

> **What is apple-app-site-association do?**
> The @apple-app-site-association file is used for Universal Links in iOS. It specifies the paths within the app that can be opened directly from a website. 

So we can just simply add the workspace details URL, and the Smart App Banner will also load on the Workspace details page, thus fixing the Visual flickering.

```
{
    "applinks": {
        "details": [
            {
                "appIDs": ["368M544MTT.com.chat.expensify.chat"],
                "components": [
                    {
                        "/": "/workspace/*",
                        "comment": "Details of Workspace" // Add this here
                    },
                    {
                       "/": "/r/*",
                       "comment": "Reports"
                    },
                    {
                        "/": "/a/*",
                        "comment": "Profile Page"
                    },
                    {
                        "/": "/settings/*",
                        "comment": "Profile and app settings"
                    },
                    {
                        "/": "/details/*",
                        "comment": "Details of another users"
                    },
                    {
                        "/": "/v/*",
                        "comment": "Account validation"
                    },
                    {
                        "/": "/bank-account/*",
                        "comment": "Bank account setup and its steps"
                    },
                    {
                        "/": "/iou/*",
                        "comment": "I Owe You reports"
                    },
                    {
                        "/": "/request/*",
                        "comment": "Money request"
                    },
                    {
                        "/": "/enable-payments/*",
                        "comment": "Payments setup"
                    },
                    {
                        "/": "/partners/plaid/oauth_ios",
                        "comment": "Plaid setup"
                    },
                    {
                        "/": "/statements/*",
                        "comment": "Wallet statements"
                    },
                    {
                        "/": "/concierge/*",
                        "comment": "Concierge"
                    },
                    {
                        "/": "/split/*",
                        "comment": "Split Bill"
                    },
                    {
                        "/": "/request/*",
                        "comment": "Request Money"
                    },
                    {
                        "/": "/new/*",
                        "comment": "New Chat"
                    }
                ]
            }
        ]
    },
    "webcredentials": {
        "apps": ["368M544MTT.com.chat.expensify.chat"]
    }
}
```

Here are some cross-checks that the URL listed shows the Smart App Banner correctly

| Related Page with URL structure | Screenshot |
|----------------------------------|------------|
| On Reports related pages: `/r/*`  | ![:r:*](https://github.com/Expensify/App/assets/20473526/660c4a20-62d7-437f-8461-a0dfaac08d7a) |
| On Settings page:`/settings`  | ![:settings](https://github.com/Expensify/App/assets/20473526/f5d33a0c-ea8f-48c7-a3e5-0997b3130b86) |
| On Settings Workspace page:`/settings/*`  | ![:settings:workspace](https://github.com/Expensify/App/assets/20473526/29361537-a0e4-45b7-a638-9e6f9b7053d0) |
| On Profiles pages:`/a/*`  | ![:a:*](https://github.com/Expensify/App/assets/20473526/95a34bf7-8e6f-4629-a8b7-51c43e8fa581) |

## TLDR @allroundexperts 
So instead of configuring Smart App Banner from meta configuration, we can just extend the list to include `/workspace/*` URL path

**Comment by studentofcoding** [comment]
All the details are listed above @allroundexperts, if all clear we can have the PR ready today

**Comment by sophiepintoraetz** [comment]
Just waiting on a go ahead from @allroundexperts

**Comment by dangrous** [comment]
Cool, I think that proposal makes sense - a couple questions:
* are there other routes that we should add details for in that file? Or is everything else covered?
* Where does the `comment` show up? Should we grab a marketing team member to get us some copy for that?

**Comment by studentofcoding** [comment]
> Cool, I think that proposal makes sense - a couple questions:
> 
> * are there other routes that we should add details for in that file? Or is everything else covered?
> * Where does the `comment` show up? Should we grab a marketing team member to get us some copy for that?

Thanks for the follow-up @allroundexperts & @dangrous 

* For the other routes, I saw that we have new Teacher pages, If the path is already stable & fixed I think we can add a `/save-the-world/*` path to manage it too, but if It's still in development, I'll suggest to do this after it's stable
* For the comment, it's not shown on the app, rather than just a standard helper for adding a comment to help us

Anyway on my side, as we're in a Merge Freeze situation (related to the React Native EU conference), will this issue be postponed or will it be prioritized to be merged @dangrous?

**Comment by dangrous** [comment]
Okay cool. I think it probably can't hurt to add save-the-world as well?

I think we can probably get this merged despite the freeze since it's unlikely to affect anything else, so I'm confirming that before assigning (since I want to make sure you still get the bonus)

**Comment by studentofcoding** [comment]
> Okay cool. I think it probably can't hurt to add save-the-world as well?
> 
> I think we can probably get this merged despite the freeze since it's unlikely to affect anything else, so I'm confirming that before assigning (since I want to make sure you still get the bonus)

Thanks for your support @dangrous, I'll standby for it.

Also, in this case, it shouldn't hurt, just share my concern about the possibility of regression on `/save-the-world/*` (as we only can see the layout change on it after being deployed & might spread the scope). But if all is good, then I'll include the `/save-the-world/*` too

**Comment by dangrous** [comment]
Okay yep we can go ahead with this! I will assign you.

As for save the world I'll just ping one of the folks who is working on that so they're aware, I don't think it'll be an issue.

**Comment by studentofcoding** [comment]
Noted thanks @dangrous, the PR is ready to be reviewed! https://github.com/Expensify/App/pull/26172

**Comment by studentofcoding** [comment]
Anyway as I've applied to Upwork and the PR is opened, could you accept my proposal @sophiepintoraetz ? Thanks

**Comment by sophiepintoraetz** [comment]
@studentofcoding - I don't have to accept it until payment is due - don't worry!

**Comment by studentofcoding** [comment]
Leave this here for a context of merging within 3 days (or on 1 Sept) but it got delayed as Merge Freeze

https://github.com/Expensify/App/pull/26172#issuecomment-1703077692
<img width="1028" alt="Screenshot 2023-09-12 at 22 07 16" src="https://github.com/Expensify/App/assets/20473526/d06cb491-ec1d-44e4-a7f0-bd06488ed010">

**Comment by studentofcoding** [comment]
Hey guys, just checked the Upwork page, and the job is already closed, even though I haven't yet received the Offer.

<img width="1552" alt="Screenshot 2023-09-13 at 16 13 28" src="https://github.com/Expensify/App/assets/20473526/c5b6bd27-bc25-4595-9a7b-dd1602036542">

Can anybody share what happened? @dangrous @sophiepintoraetz 

Thanks

**Comment by sophiepintoraetz** [comment]
That happens when the job posting is too old.

**Comment by studentofcoding** [comment]
> That happens when the job posting is too old but I can still accept proposals (edit) I'll need to look into this.

Noted, thanks for speedy follow-up @sophiepintoraetz

**Comment by studentofcoding** [comment]
## Follow up

Hey guys, just following up on this issue, after the last deployment on **Staging (v1.3.70-2)** Smart App Banner is finally showing on WS details!

### Staging (v1.3.70-2)
https://github.com/Expensify/App/assets/20473526/f268cce4-c9e7-4c5e-b5f8-9c2a0dd362a9

But a weird & interesting quirk is happening on Production (v1.3.69-2) as the Smart App Banner didn't show

### Production (v1.3.69-2
https://github.com/Expensify/App/assets/20473526/3173b46b-a22d-49c7-b499-9b8c0424e9e9

This behavior is really interesting, as we have the same structure for `/workspace/*` URL on `apple-app-site-association` & the only differences are on the updated `/teachersunite/*`that being updated by Internal PR [here](https://github.com/Expensify/App/commit/10c41da000bb36d46da014111921b04ddadae50d)

| Staging | Production |
| ---- | ---- |
| <img width="1552" alt="staging" src="https://github.com/Expensify/App/assets/20473526/3a97bd78-8999-42cf-aaa1-16b895e9f4e1"> | <img width="1552" alt="production" src="https://github.com/Expensify/App/assets/20473526/09721ab8-73c2-4074-9a4a-64d96f83762f"> |

My hunch is this behavior is happening because the Production still didn't have the last updated change from Staging, which is the change from internal PR [below](https://github.com/Expensify/App/commit/10c41da000bb36d46da014111921b04ddadae50d) (that basically change `/save-the-world/*` into `/teachersunite/*` that make the Smart App Banner didn't behave as expected

What do you think of this guy's @dangrous @allroundexperts? Just want to follow this up, so if there's another step needed from my side I'll gladly add it before the final Production build :)

Also, could I ask for your insight @aswin-s ? thank you!

**Comment by aswin-s** [comment]
@studentofcoding Everything looks good to me. We'll just have to wait for `v1.3.70-2` to be deployed to Prod. Smart banner is shown only when the installed binary supports the url prefix of the current web page and the corresponding entry is added to `apple-app-site-association`.

**Comment by studentofcoding** [comment]
> @studentofcoding Everything looks good to me. We'll just have to wait for `v1.3.70-2` to be deployed to Prod. Smart banner is shown only when the installed binary supports the url prefix of the current web page and the corresponding entry is added to `apple-app-site-association`. 

Hey @aswin-s, thank you for your confirmation & sharing the detailed process!

**Comment by studentofcoding** [comment]
Hey guys, I saw that @MelvinBot, didn't mention any automatic process on merging to main or staging nor regression period, Is this normal? Thanks!

cc: @dangrous @sophiepintoraetz

**Comment by dangrous** [comment]
He's been acting up lately, haha. This is merged, on staging, not quite on prod, but almost there!

**Comment by studentofcoding** [comment]
🤣 Thanks @dangrous! 

In the mean time, do I need to re-applied to Upwork @sophiepintoraetz? Thanks

**Comment by studentofcoding** [comment]
> @studentofcoding Seems to be working for me? 
> 
> https://github.com/Expensify/App/assets/6880914/f3d62254-9226-4b80-92a4-384f3c9424aa

Thanks for your confirmation @aswin-s !

It's finally working on my end too after remove the cache and safari data and (re-install the App)

https://github.com/Expensify/App/assets/20473526/7a96a9be-212f-4013-aa55-e0d7bb9e9964

**Comment by dangrous** [comment]
Just doing Melvin's job for him (🙄) this was deployed to production 18 hours ago, so we can handle payment (assuming no regressions) **September 25th 6:13PM GMT+8**!

**Comment by studentofcoding** [comment]
> Just doing Melvin's job for him (🙄) this was deployed to production 18 hours ago, so we can handle payment (assuming no regressions) **September 25th 6:13PM GMT+8**!

What a lazy Melvin! 😂

Thanks for this @dangrous !

**Comment by sophiepintoraetz** [comment]
@studentofcoding - here's [the new job](https://www.upwork.com/jobs/~015e5384c9955992e4) for you to apply to. It would help if you referenced your Upworks profile or GH handle so we can identify you and hire you directly.

@allroundexperts - I think you can raise a ND request for $1000 (potentially including the bonus if there are no regressions that arise).

**Comment by studentofcoding** [comment]
> @studentofcoding - here's [the new job](https://www.upwork.com/jobs/~015e5384c9955992e4) for you to apply to. It would help if you referenced your Upworks profile or GH handle so we can identify you and hire you directly.
> 
> 
> 
> @allroundexperts - I think you can raise a ND request for $1000 (potentially including the bonus if there are no regressions that arise).

Sure, the Upwork and GH handle are sended also within the Proposal @sophiepintoraetz

**Comment by sophiepintoraetz** [comment]
>so that it does not block my current payment.

Oh interesting, why would it do that??

**Comment by studentofcoding** [comment]
> @studentofcoding - here's [the new job](https://www.upwork.com/jobs/~015e5384c9955992e4) for you to apply to. It would help if you referenced your Upworks profile or GH handle so we can identify you and hire you directly.
> 
> @allroundexperts - I think you can raise a ND request for $1000 (potentially including the bonus if there are no regressions that arise).

Anyway, I registered the GH and Upwork profile on another Issue a week ago too @sophiepintoraetz. But if it's still needed here, here are the details:

Contributor details
Your Expensify account email: [yonathanevan@gmail.com](mailto:yonathanevan@gmail.com)
Upwork Profile Link: https://www.upwork.com/freelancers/~01ca6ccb72fa9cc678

On another note, I think Upwork Job is still not accepted, thanks!

**Comment by studentofcoding** [comment]
Contributor details
Your Expensify account email: [yonathanevan@gmail.com](mailto:yonathanevan@gmail.com)
Upwork Profile Link: https://www.upwork.com/freelancers/~01ca6ccb72fa9cc678

**Comment by studentofcoding** [comment]
> Leave this here for a context of merging within 3 days (or on 1 Sept) but it got delayed as Merge Freeze
> 
> [#26172 (comment)](https://github.com/Expensify/App/pull/26172#issuecomment-1703077692) <img alt="Screenshot 2023-09-12 at 22 07 16" width="1028" src="https://user-images.githubusercontent.com/20473526/267375938-d06cb491-ec1d-44e4-a7f0-bd06488ed010.png">

Share it again here

cc: @sophiepintoraetz

**Comment by studentofcoding** [comment]
> Leave this here for a context of merging within 3 days (or on 1 Sept) but it got delayed as Merge Freeze
> 
> [#26172 (comment)](https://github.com/Expensify/App/pull/26172#issuecomment-1703077692) <img alt="Screenshot 2023-09-12 at 22 07 16" width="1028" src="https://user-images.githubusercontent.com/20473526/267375938-d06cb491-ec1d-44e4-a7f0-bd06488ed010.png">

Looking forward for the Job acceptance on Upwork and Payment @sophiepintoraetz

Also, Share the context above, as it's on 50% Urgency bonuses 

cc: @dangrous @allroundexperts

**Comment by sophiepintoraetz** [comment]
Payouts due:

Issue Reporter: N/A (Applause)
Contributor: $1500 @studentofcoding 
Contributor+: $1500 @allroundexperts 

Eligible for 50% #urgency bonus? Y

**Comment by JmillsExpensify** [comment]
$1,500 payment approved for @allroundexperts based on BZ summary.
